app/javascript/components/student/solution-summary/Mentoring.tsx

```diff
@@ -10,16 +28,115 @@ export const Mentoring = ({ link }: { link: string }): JSX.Element => {
         On average, students iterate a further 3.5 times when mentored on a
         solution.
       </p>
-      <div className="c-combo-button">
-        <button className="--editor-segment">Request Mentoring</button>
-        <button className="--dropdown-segment">
-          <GraphicalIcon icon="chevron-down" />
-        </button>
-      </div>
-      <a href={link} className="learn-more">
+      <MentoringComboButton
+        hasMentorDiscussionInProgress={hasMentorDiscussionInProgress}
+        discussions={discussions}
+        links={links}
+      />
+      <a href={links.learnMoreAboutMentoringArticle} className="learn-more">
         Learn more
         <Icon icon="external-link" alt="Opens in new tab" />
       </a>
     </div>
   )
 }
+
+const MentoringComboButton = ({
+  hasMentorDiscussionInProgress,
+  discussions,
+  links,
+}: {
+  hasMentorDiscussionInProgress: boolean
+  discussions: readonly MentorDiscussion[]
+  links: {
+    requestMentoring: string
+    shareMentoring: string
+  }
+}) => {
+  const {
+    open,
+    setOpen,
+    setButtonElement,
+    setPanelElement,
+    styles,
+    attributes,
+  } = usePanel({
+    placement: 'bottom',
+    modifiers: [
+      {
+        name: 'offset',
+        options: {
+          offset: [0, 14],
+        },
+      },
+    ],
+  })
+
+  return (
+    /* TODO: Extract into a common component in the future */
+    <div className="c-combo-button">
+      {discussions.length > 0 ? (
+        <a href={discussions[0].links.self} className="--editor-segment">
+          Continue mentoring
+        </a>
+      ) : (
+        <a href={links.requestMentoring} className="--editor-segment">
+          Request mentoring
+        </a>
+      )}
+      <button
+        className="--dropdown-segment"
+        onClick={() => {
+          setOpen(!open)
+        }}
+        ref={setButtonElement}
+      >
+        <GraphicalIcon icon="chevron-down" />
+      </button>
+      <MentoringPanel
+        setPanelElement={setPanelElement}
+        open={open}
+        styles={styles}
+        attributes={attributes}
+      >
+        <MentoringDropdown
+          hasMentorDiscussionInProgress={hasMentorDiscussionInProgress}
+          discussions={discussions}
+          links={{ share: links.shareMentoring }}
+        />
+      </MentoringPanel>
+    </div>
+  )
+}
+
+type PanelProps = {
+  open: boolean
+  styles: { [key: string]: React.CSSProperties }
+  attributes: {
+    [key: string]: {
+      [key: string]: string
+    }
+  }
+  setPanelElement: React.Dispatch<React.SetStateAction<HTMLDivElement | null>>
+}
+
+const MentoringPanel = ({
```

Comment:
I render the dropdown panel in a portal in order for it to not inherit the styles from the parent div. We should probably do this for all of our panels moving forward.

Reply:
I'm not entirely sure. I'll look into it.